### ISSUE TITLE
Build errors on CentOS 6.7 

### ISSUE THREAD

**Issue by remusvrm** [OpenImageIO/oiio#1406]
### Problem

I am trying to compile OpenImageIO Release-1.6.12 on CentOS 6.7, but unfortunately I get this error:

```
.........
[ 14%] Building CXX object src/libutil/CMakeFiles/ustring_test.dir/ustring_test.cpp.o
.../openimageIO/1.6.12/src/oiio-Release-1.6.12/src/libutil/simd_test.cpp: In function ‘void test_component_access() [with VEC = OpenImageIO::v1_6::simd::mask4]’:
.../openimageIO/1.6.12/src/oiio-Release-1.6.12/src/libutil/simd_test.cpp:175: error: using ‘typename’ outside of template
make[2]: *** [src/libutil/CMakeFiles/simd_test.dir/simd_test.cpp.o] Error 1
make[1]: *** [src/libutil/CMakeFiles/simd_test.dir/all] Error 2
```

### Versions

* OIIO branch/version:  oiio-Release-1.6.12
* OS:  CentOS 6.7
* C++ compiler: gcc version 4.4.7 20120313 (Red Hat 4.4.7-16) (GCC)
* PYTHON_VERSION='2.7'
* BOOST_VERSION='1.44.0' (static libs)
* CMAKE_VERSION='3.3.2'
* OPENEXR_VERSION='2.2.0'
* LIBJPEGTURBO_VERSION='1.4.2-1'
* OPENCOLORIO_VERSION='1.0.7'

### Configure
```
cd ${PACKAGE_NAME}
mkdir build
cd build
cmake -DILMBASE_HOME=${ILMBASE_HOME} \
      -DILMBASE_INCLUDE_DIR=${ILMBASE_INCLUDE_DIR} \
      -DOPENEXR_HOME=${OPENEXR_HOME} \
      -DOPENEXR_INCLUDE_DIR=${OPENEXR_INCLUDE_DIR} \
      -DOPENEXR_ILMIMF_LIBRARIES=${OPENEXR_ILMIMF_LIBRARIES} \
      -DBOOST_ROOT=${BOOST_ROOT} \
      -fPIC \
      -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_DIR} \
      -DUSE_OPENJPEG=1 \
      -DJPEG_INCLUDE_DIR=${JPEG_INCLUDE_DIR} \
      -DOCIO_PATH=${OPENCOLORIO_PATH} \
      -DUSE_PYTHON=1 \
      -DPYTHON_VERSION=2.7 \
      -DUSE_FFMPEG=1 \
      -DUSE_GIF=1 \
      -DUSE_LIBRAW=1 \
      ..
```

**Comment by remusvrm** [comment]
I compiled OpenImageIO Release-1.6.11 on CentOS 6.7 without any errors.

**Comment by lgritz** [comment]
I think I see what's going on, will have a fix for you today.

**Comment by lgritz** [comment]
Proposed fix in #1407

**Comment by remusvrm** [comment]
Thank you Igritz! I will test it and let you know if still doesn't work.
